interfaces.go

```diff
@@ -0,0 +1,215 @@
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/linode/linodego/internal/parseabletime"
+)
+
+type LinodeInterface struct {
+	ID           int                     `json:"id"`
+	Version      int                     `json:"version"`
+	MACAddress   string                  `json:"mac_address"`
+	Created      *time.Time              `json:"-"`
+	Updated      *time.Time              `json:"-"`
+	DefaultRoute *InterfacesDefaultRoute `json:"default_route,omitempty"`
+	Public       *InterfacesPublic       `json:"public,omitempty"`
+	VPC          *InterfacesVPC          `json:"vpc,omitempty"`
+	VLAN         *InterfacesVLAN         `json:"vlan,omitempty"`
+}
+
+type InterfacesDefaultRoute struct {
+	IPv4 *bool `json:"ipv4,omitempty"`
+	IPv6 *bool `json:"ipv6,omitempty"`
+}
+
+type InterfacesPublic struct {
+	IPv4 *InterfacesIPv4 `json:"ipv4,omitempty"`
+	IPv6 *InterfacesIPv6 `json:"ipv6,omitempty"`
+}
+
+type InterfacesIPv4 struct {
+	Addresses []InterfacesIPv4Address `json:"addresses"`
+	Shared    []InterfacesIPv4Shared  `json:"shared"`
+}
+
+type InterfacesIPv6 struct {
+	Ranges []InterfacesIPv6Range `json:"ranges"`
+	Shared []InterfacesIPv6Range `json:"shared"`
+	Slaac  []InterfacesIPv6Slaac `json:"slaac"`
+}
+
+type InterfacesIPv4Address struct {
+	Address string `json:"address"`
+	Primary bool   `json:"primary"`
+}
+
+type InterfacesIPv4Shared struct {
+	Address  string `json:"address"`
+	LinodeID string `json:"linode_id"`
+}
+
+type InterfacesIPv6Range struct {
+	Range       string `json:"range"`
+	RouteTarget string `json:"route_target"`
+}
+
+type InterfacesIPv6Slaac struct {
+	Prefix  int    `json:"prefix"`
+	Address string `json:"address"`
+}
+
+type InterfacesVPC struct {
+	VPCID    int               `json:"vpc_id"`
+	SubnetID int               `json:"subnet_id"`
+	IPv4     InterfacesVPCIPv4 `json:"ipv4"`
+}
+
+type InterfacesVPCIPv4 struct {
+	Addresses []InterfacesVPCIPv4Address `json:"addresses"`
+	Ranges    []InterfacesVPCIPv4Range   `json:"ranges"`
+}
+
+type InterfacesVPCIPv4Address struct {
+	Address        string  `json:"address"`
+	Primary        bool    `json:"primary"`
+	NAT1to1Address *string `json:"nat_1_1_address,omitempty"`
+}
+
+type InterfacesVPCIPv4Range struct {
+	Range string `json:"range"`
+}
+
+type InterfacesVLAN struct {
+	Label       string  `json:"vlan_label"`
+	IPAMAddress *string `json:"ipam_address,omitempty"`
+}
+
+type LinodeInterfaceCreateOptions struct {
+	FirewallID   *int                           `json:"firewall_id,omitempty"`
+	DefaultRoute *InterfacesDefaultRoute        `json:"default_route,omitempty"`
+	Public       *InterfacesPublicCreateOptions `json:"public,omitempty"`
+	VPC          *InterfacesVPCCreateOptions    `json:"vpc,omitempty"`
+	VLAN         *InterfacesVLAN                `json:"vlan,omitempty"`
+}
+
+type InterfacesPublicCreateOptions struct {
+	IPv4 []*InterfacesIPv4              `json:"ipv4,omitempty"`
+	IPv6 []*InterfacesIPv6CreateOptions `json:"ipv6,omitempty"`
```

Comment:
```suggestion
	IPv4 []InterfacesIPv4              `json:"ipv4,omitempty"`
	IPv6 []InterfacesIPv6CreateOptions `json:"ipv6,omitempty"`
```
A similar suggestion to the previous one.